tdm-reader/nitdmreader.yml

```diff
@@ -0,0 +1,512 @@
+      x-ni-request-all-privileges: true
+      security: []
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/V1Operations'
+        404:
+          description: Not Found
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/files/{file-id}:
+    get:
+      tags: [metadata]
+      summary: Gets file, group, and channel information
+      description: 'Gets all group names, channel names, channel information, and properties at all levels. This does not return channel data.'
+      operationId: getAllFileInfo
+      produces:
+        - application/json
+      parameters:
+        - name: file-id
+          description: File id
+          in: path
+          type: string
+          required: true
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/FileInformation'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/query-files:
+    post:
+      tags: [metadata]
+      summary: Gets file, group, and channel information
+      description: 'Gets all group names, channel names, channel information, and properties at all levels. This does not return channel data.'
+      operationId: getMultipleFileInfo
+      produces:
+        - application/json
+      parameters:
+        - name: fileIds
+          description: List of File Ids to inspect.
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/FileList'
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/FileInformation'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/files/{file-id}/groups:
+    get:
+      tags: [metadata]
+      summary: Lists all groups in the file
+      operationId: getChannelPropertiesCount
+      produces:
+        - application/json
+      parameters:
+        - name: file-id
+          description: File id
+          in: path
+          required: true
+          type: string
+        - name: skip
+          description: Number of items to skip
+          in: query
+          type: integer
+          default: 0
+        - name: take
+          description: Number of items to take after skipping
+          in: query
+          type: integer
+          default: -1
+        - name: returnCount
+          description: Specify this to return the total count of items
+          in: query
+          type: boolean
+          default: false
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/NamesList'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/files/{file-id}/groups/{group-name}/channels:
+    get:
+      tags: [metadata]
+      summary: Lists all channels in the group
```

Comment:
Other Swagger documents I've reviewed have both a summary and description. Is there a specific reason to only have summary? I don't know a ton about implementation. Could maybe ask Phil about this.